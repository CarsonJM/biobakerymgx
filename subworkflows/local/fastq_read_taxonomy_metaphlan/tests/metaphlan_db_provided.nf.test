nextflow_workflow {

    name "Test Subworkflow: FASTQ_READ_TAXONOMY_METAPHLAN"
    script "../main.nf"
    workflow "FASTQ_READ_TAXONOMY_METAPHLAN"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_read_taxonomy_metaphlan"
    tag "fastq_read_taxonomy_metaphlan_db_provided"


    test(" metaphlan_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] ") {

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir          = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test1' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = "https://github.com/biobakery/MetaPhlAn/raw/master/metaphlan/utils/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB2GTDB.tsv"
                input[2] = Channel.of(
                    params.metaphlan_db
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test(" metaphlan_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] - stub") {

        options "-stub"

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir          = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test1' ],
                        [
                            file(params.test_data['bacteroides_fragilis']['illumina']['test1_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['bacteroides_fragilis']['illumina']['test1_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = "https://github.com/biobakery/MetaPhlAn/raw/master/metaphlan/utils/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB2GTDB.tsv"
                input[2] = Channel.of(
                    params.metaphlan_db
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
