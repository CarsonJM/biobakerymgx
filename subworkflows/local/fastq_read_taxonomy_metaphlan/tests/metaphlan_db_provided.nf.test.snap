{
    " metaphlan_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d70905cc040742d5e72b3a037888ee00"
                    ]
                ],
                "1": [
                    "versions.yml:md5,932a88b79ff4ac88dea7e3ea0b85af9b",
                    "versions.yml:md5,96c5194c76c69ebe464df7904e44de26",
                    "versions.yml:md5,ea68a854ad5e92799f15dbcf59f3795f"
                ],
                "metaphlan_profiles_merged_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d70905cc040742d5e72b3a037888ee00"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,932a88b79ff4ac88dea7e3ea0b85af9b",
                    "versions.yml:md5,96c5194c76c69ebe464df7904e44de26",
                    "versions.yml:md5,ea68a854ad5e92799f15dbcf59f3795f"
                ]
            }
        ],
        "timestamp": "2024-01-24T10:13:44.282677237"
    },
    " metaphlan_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,932a88b79ff4ac88dea7e3ea0b85af9b",
                    "versions.yml:md5,96c5194c76c69ebe464df7904e44de26",
                    "versions.yml:md5,ea68a854ad5e92799f15dbcf59f3795f"
                ],
                "metaphlan_profiles_merged_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,932a88b79ff4ac88dea7e3ea0b85af9b",
                    "versions.yml:md5,96c5194c76c69ebe464df7904e44de26",
                    "versions.yml:md5,ea68a854ad5e92799f15dbcf59f3795f"
                ]
            }
        ],
        "timestamp": "2024-01-24T10:14:16.478201535"
    }
}